node/src/components/block_validator.rs

```diff
@@ -216,9 +217,98 @@ where
                         true
                     }
                 });
+            }
+            Event::Loaded { .. } => {}
+        }
+        effects
+    }
+}
 
-                effects
+/// Block validator.
+#[derive(Debug)]
+pub(crate) enum BlockValidatorState<T, I> {
+    Loading(Vec<Event<T, I>>),
+    Ready(BlockValidatorReady<T, I>),
+}
+
+/// Block validator.
+#[derive(DataSize, Debug)]
+pub(crate) struct BlockValidator<T, I> {
+    #[data_size(skip)]
+    state: BlockValidatorState<T, I>,
+    _marker: std::marker::PhantomData<I>,
```

Comment:
This can probably be removed now I think.

Reply:
Yep, thanks.